test/gtest/bn_fwd_infer_ocl_hip.cpp

```diff
@@ -0,0 +1,274 @@
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <miopen/env.hpp>
+#include <miopen/miopen.h>
+#include <miopen/kernel_build_params.hpp>
+
+#include <miopen/batchnorm/problem_description.hpp>
+
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include <miopen/solver/ck_utility_common.hpp>
+
+#include "bn_test_data.hpp"
+#include "test_operations.hpp"
+
+void BatchNormForwardInferencGPU(miopen::Handle& handle,
+                                 miopenBatchNormMode_t bn_mode,
+                                 const void* alpha,
+                                 const void* beta,
+                                 const miopen::TensorDescriptor& xDesc,
+                                 ConstData_t x,
+                                 const miopen::TensorDescriptor& yDesc,
+                                 Data_t y,
+                                 const miopen::TensorDescriptor& bnScaleBiasMeanVarDesc,
+                                 ConstData_t bnScale,
+                                 ConstData_t bnBias,
+                                 ConstData_t estimatedMean,
+                                 ConstData_t estimatedVariance,
+                                 double epsilon,
+                                 bool use_hip = false)
+{
+
+    bool bfpmixparm = false;
+    bool bfp16parm  = false;
+    bool bfp32parm  = true;
+
+    if(xDesc.GetType() == miopenHalf && bnScaleBiasMeanVarDesc.GetType() == miopenHalf)
+    {
+        bfp16parm = true;
+        bfp32parm = false;
+    }
+    else if(xDesc.GetType() == miopenHalf && bnScaleBiasMeanVarDesc.GetType() == miopenFloat)
+    {
+        bfpmixparm = true;
+        bfp32parm  = false;
+    }
+
+    int n, c, h, w;
+    std::tie(n, c, h, w) = miopen::tien<4>(xDesc.GetLengths());
```

Comment:
```suggestion
    auto [n, c, h, w] = miopen::tien<4>(xDesc.GetLengths());
```
C++17 has a special structure binding syntax. It works perfect with simple values like `ints` or `floats`.

Reply:
GetLengths() returns size_t, but I expect n,c,h,w to be ints. So, sticking to doing whatever I was doing before.